### PR TITLE
[PAYG] [dashboard] Onboarding: Product Educational content in `/ workspaces` view

### PR DESCRIPTION
**Pull request by Siddhant-K-code**
## Description

This PR adds educational content to the dashboard.

## Related Issue(s)

Fixes OPM-657

## How to test

- Open the preview environment & complete the signup process
- Create a new workspace and then see on the right side of the `/workspaces` page 👀

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - siddhant-of04b0355e5</li>
	<li><b>🔗 URL</b> - <a href="https://siddhant-of04b0355e5.preview.gitpod-dev.com/workspaces" target="_blank">siddhant-of04b0355e5.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - siddhant-opm-657-add-educational-information-to-the-dashboard-video-links-gha.27318</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-siddhant-of04b0355e5%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-dev-preview" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [x] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/dashboard/src/workspaces/BlogBanners.tsx
Comment (Siddhant-K-code) [inline components/dashboard/src/workspaces/BlogBanners.tsx:39]: It introduces a new key-value pair in local storage to check the index for blog banners (which would be switched in every 2 days amongst these 4s: 

https://github.com/gitpod-io/gitpod/blob/c9fc1c99a38d2455b4df21ef2c927c1b49d2bdb0/components/dashboard/src/workspaces/BlogBanners.tsx#L10-L31

File: components/dashboard/src/workspaces/Workspaces.tsx
Comment (filiptronicek) [inline components/dashboard/src/workspaces/Workspaces.tsx:182]: Is there no way to dismiss this side banner? It feels really intrusive considering there's bound to be many users who either already are knowledgeable about these topics or at least know our docs are also accessible via the account dropdown.

File: components/dashboard/src/workspaces/Workspaces.tsx
Comment (svenefftinge) [inline components/dashboard/src/workspaces/Workspaces.tsx:110]: `className="flex-1"` will make this part use up the space

File: components/dashboard/src/workspaces/VideoCarousel.tsx
Comment (filiptronicek) [inline components/dashboard/src/workspaces/VideoCarousel.tsx:65]: These buttons lack focus rings / indicators, making it hard to discern which one is to-be selected.

File: components/dashboard/src/workspaces/EmptyWorkspacesContent.tsx
Comment (filiptronicek) [inline components/dashboard/src/workspaces/EmptyWorkspacesContent.tsx:18]: thanks for removing this, forgot to in my PR from yesterday 🙏

File: components/dashboard/src/workspaces/Workspaces.tsx
Comment (filiptronicek) [inline components/dashboard/src/workspaces/Workspaces.tsx:92]: nit: explicitly mentioning starting and stopping seems redundant if we're already saying you can manage them.

File: components/dashboard/src/workspaces/VideoCarousel.tsx
Comment (filiptronicek) [inline components/dashboard/src/workspaces/VideoCarousel.tsx:65]: nit: it's somewhat confusing to switch between videos with just these ellipses. If I wanted to find a video among these, I would have to go through each one.

**Review by filiptronicek — COMMENTED** [review]
changes look just like the designs, great work! I left a couple of comments

File: components/dashboard/src/workspaces/VideoCarousel.tsx
Comment (Siddhant-K-code) [reply components/dashboard/src/workspaces/VideoCarousel.tsx:66]: Selected button shows as primary kumquat colour and on hover It changes to this (second dot)

image

Comment (Siddhant-K-code) [reply components/dashboard/src/workspaces/VideoCarousel.tsx:66]: What alternative UX design would you suggest to fix this?

File: components/dashboard/src/workspaces/Workspaces.tsx
Comment (Siddhant-K-code) [reply components/dashboard/src/workspaces/Workspaces.tsx:92]: I've picked it from Figma Designs, will check it once with @phimae

File: components/dashboard/src/workspaces/Workspaces.tsx
Comment (Siddhant-K-code) [reply components/dashboard/src/workspaces/Workspaces.tsx:182]: Check this [slack message](https://gitpod.slack.com/archives/C074K0JASMQ/p1720545416829909) for context 🙏🏻

File: components/dashboard/src/workspaces/Workspaces.tsx
Comment (Siddhant-K-code) [reply components/dashboard/src/workspaces/Workspaces.tsx:110]: Resolved in [`e7e8772`](https://github.com/gitpod-io/gitpod/pull/20021/commits/e7e8772a7f7a3c535345d8884c63947a579574fd)

File: components/dashboard/src/workspaces/Workspaces.tsx
Comment (Siddhant-K-code) [reply components/dashboard/src/workspaces/Workspaces.tsx]: Looks like it was a old one (which was outdated in Figma), reverted this change in [`7008c69`](https://github.com/gitpod-io/gitpod/pull/20021/commits/7008c696da75a5cb82cddd3e9e3da3194de53b69)

File: components/dashboard/src/workspaces/VideoCarousel.tsx
Comment (filiptronicek) [reply components/dashboard/src/workspaces/VideoCarousel.tsx:66]: Sorry for the misunderstanding: the focus rings are missing for keyboard navigation, when tabbing through the buttons to select a video.

Comment (filiptronicek) [reply components/dashboard/src/workspaces/VideoCarousel.tsx:66]: I think we would have to reconsider this entire section (a proper and informed carousel wouldn't fit), where we'd have the video title or a keyword relevant to the video visible for every one. Most likely not the effort at this point, though

Comment (Siddhant-K-code) [reply components/dashboard/src/workspaces/VideoCarousel.tsx:66]: Yeah, that would be good to have. But, I won’t consider this as a blocker for this PR. Maybe something for a follow-up?

**Comment by loujaybee** [comment]
One final minor design tweak: 
1. Increase the spacing between the elements (above each header for docs, personalized and above the image)
2. Reduce the spacing under the headers and the list of links
3. Reduce the spacing between the links

It feels a bit visually tight and adding this spacing makes it visually easier to see which elements are grouped. 

|Before|After|
|-|-|
|image|image|

**Review by loujaybee — APPROVED** [review]

File: components/dashboard/src/workspaces/VideoCarousel.tsx
Comment (filiptronicek) [inline components/dashboard/src/workspaces/VideoCarousel.tsx:51]: Because we're mapping over videos here and using the global state `currentVideo`, we will end up with many `<lite-youtube>` with the same key. We should ensure those are [unique](https://react.dev/learn/rendering-lists#keeping-list-items-in-order-with-key) and because the parent already has a unique key, we should instead just either just our `Tabs` component, or dynamically render the currently selected video and keep the rest unmnounted.

Comment (Siddhant-K-code) [reply components/dashboard/src/workspaces/VideoCarousel.tsx:51]:
```tsx
key = {video.id}
```

was leading to that bug 🐛, in which it kept getting played if we even changed the video (clicked on dot 2)

**Comment by Siddhant-K-code** [comment]
@loujaybee I've resolved [your comment](https://github.com/gitpod-io/gitpod/pull/20021#issuecomment-2222984975) in [`1ea3a32`](https://github.com/gitpod-io/gitpod/pull/20021/commits/1ea3a32efea91c68e4f6cbb51f976ce75f51d131)

File: components/dashboard/src/workspaces/VideoCarousel.tsx
Comment (phimae) [reply components/dashboard/src/workspaces/VideoCarousel.tsx:66]: Thanks Filip! Can be a follow up issue, I don't think it's a major problem right now that should block the release

File: components/dashboard/src/workspaces/VideoCarousel.tsx
Comment (filiptronicek) [reply components/dashboard/src/workspaces/VideoCarousel.tsx:51]: I understand the bug happened while the key was set to the actual video id, but we are very much doing two things react does not expect us to:
1. Use non-static `key` props
2. Have multiple `key` props have the same values

As mentioned, if we want to not have the videos continue playing, we should ditch the `.map` over `videos` and only render `videos[currentVideo]`.

File: components/dashboard/src/workspaces/VideoCarousel.tsx
Comment (filiptronicek) [reply components/dashboard/src/workspaces/VideoCarousel.tsx:66]: Agree it's no blocker 👍

File: components/dashboard/src/workspaces/VideoCarousel.tsx
Comment (Siddhant-K-code) [reply components/dashboard/src/workspaces/VideoCarousel.tsx]: Resolved in [`12ac117`](https://github.com/gitpod-io/gitpod/pull/20021/commits/12ac117ae2b9e3789e1fd251ef3407ca21d85ece)

File: components/dashboard/src/workspaces/VideoCarousel.tsx
Comment (Siddhant-K-code) [reply components/dashboard/src/workspaces/VideoCarousel.tsx:67]: Resolved in [`caa2cd1`](https://github.com/gitpod-io/gitpod/pull/20021/commits/caa2cd1ff19da80c228fb7e143b43b72a3f808f0)

Comment (filiptronicek) [reply components/dashboard/src/workspaces/VideoCarousel.tsx:67]: It's indeed not a blocker, but I think this might actually be solved by simply removing the `focus:outline-none` on the trigger buttons on line 69. It might be a real letdown for folks who rely on the keyboard when using Gitpod, I say let's include them

**Review by filiptronicek — APPROVED** [review]
looks good, approving to unblock

File: components/dashboard/src/workspaces/VideoCarousel.tsx
Comment (Siddhant-K-code) [reply components/dashboard/src/workspaces/VideoCarousel.tsx:67]: Yeah, I misunderstood your comment, it's already resolved in with our themed colour :)

**Pull request merged**
